### PR TITLE
[DataGridPro] Fix pinned columns in RTL mode

### LINKED ISSUE #14245: [data grid] Pinned columns are broken in RTL mode
### Steps to reproduce

Link to live example: [The example in the docs is also broken](https://mui.com/x/react-data-grid/column-pinning/)

Steps:
1. Create a data grid with pinned column(s)
2. Set the theme direction to RTL
3. Observe pinned column(s) do not work

### Current behavior

The pinned column does not stick to the side and some columns become invisible.
This video demonstrates the bug: 
[Screencast from 2024-08-17 13-41-52.webm](https://github.com/user-attachments/assets/57445258-c93f-4fd9-8b49-5c71af72f5c7)

### Expected behavior

Pinned columns in RTL should work like they do in LTR direction.

### Context

_No response_

### Your environment

  System:
    OS: Linux 6.10 Arch Linux
  Binaries:
    Node: 22.6.0 - /usr/bin/node
    npm: 10.8.2 - /usr/bin/npm
    pnpm: 9.7.1 - /usr/bin/pnpm
  Browsers:
    Chrome: Not Found
  npmPackages:
    @emotion/react: ^11.13.0 => 11.13.0 
    @emotion/styled: ^11.13.0 => 11.13.0 
    @mui/icons-material: ^5.16.7 => 5.16.7 
    @mui/lab: 5.0.0-alpha.169 => 5.0.0-alpha.169 
    @mui/material: ^5.16.7 => 5.16.7 
    @mui/styles: ^5.16.7 => 5.16.7 
    @mui/x-data-grid-premium: ^7.12.1 => 7.12.1 
    @mui/x-data-grid-pro: ^7.12.1 => 7.12.1 
    @mui/x-date-pickers: ^7.12.1 => 7.12.1 
    @mui/x-date-pickers-pro: ^7.12.1 => 7.12.1 
    @mui/x-license: ^7.13.0 => 7.13.0 
    @mui/x-tree-view: ^7.12.1 => 7.12.1 
    @types/react: ^18.3.3 => 18.3.3 
    react: ^18.3.1 => 18.3.1 
    react-dom: ^18.3.1 => 18.3.1 
    styled-components: ^6.1.12 => 6.1.12 
    typescript: ~5.3.3 => 5.3.3 

**Search keywords**: pinned columns,pinned,rtl

### PR DESCRIPTION
**Pull request by KenanYusuf**
Fixes #14245

**Before**: [codesandbox](https://codesandbox.io/p/sandbox/14586-rtl-pinned-columns-after-forked-2vy9l4)

Screenshot 2024-09-12 at 09 53 40

**After**: [codesandbox](https://codesandbox.io/p/sandbox/14586-rtl-pinned-columns-after-3m722n)

Screenshot 2024-09-12 at 09 53 56

**Note:** Resizing left pinned columns in RTL mode has the same issue that LTR see on right pinned columns (https://github.com/mui/mui-x/issues/12852). This PR does not attempt to fix that, as there is already a PR in progress for this issue (https://github.com/mui/mui-x/pull/12979).

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/x-data-grid/src/utils/domUtils.ts [hunk 1]
```diff
@@ -166,23 +166,31 @@ const findPinnedCells = ({
...
+  isRtl: boolean,
+) {
   const colIndex = parseCellColIndex(col);
   return findPinnedCells({
     api,
     colIndex,
-    position: 'left',
-    filterFn: (index) => index > colIndex!,
+    position: isRtl ? 'right' : 'left',
+    filterFn: (index) => (isRtl ? index < colIndex! : index > colIndex!),
```
Comment (KenanYusuf) [inline packages/x-data-grid/src/utils/domUtils.ts:179]: Inverting the position and filter function fixes an issue where the other cells within the pinned columns were not being resized:

https://github.com/user-attachments/assets/ee4096bf-74af-4f22-8209-67f1372d38e6

File: packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx [hunk 2]
```diff
@@ -939,14 +938,11 @@ export function areRenderContextsEqual(context1: GridRenderContext, context2: Gr
 export function computeOffsetLeft(
   columnPositions: number[],
   renderContext: GridColumnsRenderContext,
-  isRtl: boolean,
   pinnedLeftLength: number,
 ) {
-  const factor = isRtl ? -1 : 1;
```
Comment (KenanYusuf) [inline packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:945]: I don't fully understand the purpose of this offset, but removing this fixes this additional space between right pinned cells and non-pinned cells:

Screenshot 2024-09-11 at 16 41 29

File: test/regressions/data-grid/DataGridRTLPinnedColumns.js
Comment (cherniavskii) [inline test/regressions/data-grid/DataGridRTLPinnedColumns.js:1]: Nice, we clearly missed this in Argos 👍🏻

**Review by MBilalShafi — APPROVED** [review]
LGTM 👍

**Comment by ranpaz** [comment]
The issue of pinning columns in RTL mode has still not been resolved (#14586). If you install "@mui/x-data-grid-pro": "^7.20.0" cleanly, without remnants from previous versions, the problem persists

**Pull request merged**